app/main.py

```diff
@@ -1,2 +1,110 @@
+import dataclasses
+from typing import Hashable, Any, Optional
+
+
+@dataclasses.dataclass
+class Node:
+    key: Hashable
+    hash_: int
+    value: Any
+
+
 class Dictionary:
-    pass
+    INITIAL_CAPACITY = 8
+    LOAD_FACTOR = 2 / 3
+
+    def __init__(self, capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = capacity
+        self.length = 0
+        self.hash_table: list[Optional[Node]] = [None] * capacity
+
+    def _get_index(self, key: Hashable) -> int:
+        hash_ = hash(key)
+        index = hash_ % self.capacity
+
+        while (
+            self.hash_table[index] is not None
+            and self.hash_table[index].key != key
+        ):
+            index += 1
+            index = index % self.capacity
+
+        return index
+
+    def _resize(self) -> None:
+        new_capacity = self.capacity * 2
+        new_hash_table = [None] * new_capacity
+        for node in self.hash_table:
+            if node is not None:
+                index = node.hash_ % new_capacity
+                while new_hash_table[index] is not None:
+                    index += 1
+                    index = index % new_capacity
+                new_hash_table[index] = node
```

Comment:
Looks like you doing the same logic as you did in your `_get_index() func ` DRY !